src/context/reducers/notificationReducer.tsx

```diff
@@ -19,31 +19,38 @@ export interface NotificationItem {
 export type NotificationState = NotificationItem[];
 
 export const initialNotificationsState: NotificationState = [];
-
-// { 'variant': 'info', 'title': 'Unread info notification title', 'message': 'This is an info notification description', 'timestamp': moment().subtract(1, 'days') },
-// { 'variant': 'warning', 'title': 'Unread warning notification title', 'message': 'This is a warning!!', 'timestamp': moment().subtract(5, 'minutes') },
-// { 'variant': 'danger', 'title': 'Unread danger notification title', 'message': 'This is dangerous...', 'timestamp': moment().subtract(1, 'hour') },
-
 interface NotificationPayload {
   [NotificationActionTypes.SEND]: {
     notifications: NotificationItem[]
+  },
+  [NotificationActionTypes.CLEAR]: {},
+  [NotificationActionTypes.REMOVE]: {
+    index: number
   }
 }
 
 export type NotificationActions = ActionMap<NotificationPayload>[
   keyof ActionMap<NotificationPayload>
 ]
 
-
 export const notificationsReducer = (
   state: NotificationState,
   action: NotificationActions
 ) => {
   switch (action.type) {
-    case NotificationActionTypes.SEND: {
-      return state.concat(action.payload.notifications).sort((a, b) => b.timestamp.diff(a.timestamp)) // earliest at the front
-    }
+    case NotificationActionTypes.SEND:
+      return state.concat([...state, ...action.payload.notifications].sort((a, b) => b.timestamp.diff(a.timestamp))); // chronological order for notifications
```

Comment:
@colinywu I believe there is a bug in this implementation.
The original state and the new notification items are concatenated twice making same notifications appear multiple time and out of order since previous notifications are concatenated with prev+new notifications sorted by time.

```suggestion
      return [...state, ...action.payload.notifications].sort((a, b) => b.timestamp.diff(a.timestamp)); // chronological order for notifications
```

Reply:
Sorry, my bad. Totally implemented one of your suggestions wrong. Great catch!